config/routes.rb

```diff
@@ -127,5 +127,15 @@
   match '/:username/groups(/p/:page)', to: 'user#groups', via: 'get', as: :show_user_groups, defaults: {page: 1}
   match '/:username/questions(/p/:page)', to: 'user#questions', via: 'get', as: :show_user_questions, defaults: {page: 1}
 
+  constraints ->(req) { req.env['warden'].authenticate?(scope: :user) &&
```

Comment:
Please merge the routes in this block with the other `:administrator` constraint above.